air/src/proof/mod.rs

```diff
@@ -94,27 +87,6 @@ impl StarkProof {
         self.context.lde_domain_size()
     }
 
-    // SECURITY LEVEL
-    // --------------------------------------------------------------------------------------------
-    /// Returns security level of this proof (in bits).
-    ///
-    /// When `conjectured` is true, conjectured security level is returned; otherwise, provable
-    /// security level is returned. Usually, the number of queries needed for provable security is
-    /// 2x - 3x higher than the number of queries needed for conjectured security at the same
-    /// security level.
-    pub fn security_level(&self, conjectured: bool) -> u32 {
-        if conjectured {
-            get_conjectured_security(
-                self.context.options(),
-                self.context.num_modulus_bits(),
-                self.lde_domain_size() as u64,
-            )
-        } else {
-            // TODO: implement provable security estimation
-            unimplemented!("proven security estimation has not been implement yet")
-        }
-    }
```

Comment:
We should not delete this method - but instead find a way to pass required hash function info into it.

Reply:
As per you suggestion, I've defined a const COLLISION_RESISTANCE parameter to the `hasher` trait:

```
pub trait Hasher {
    const COLLISION_RESISTANCE: u32;
    ...
}
```

I've then defined a collision resistance parameter in `StarkProof` struct which we can use to get the security level of the hash function being used in the proof:
```
pub struct StarkProof {
    /// Collision resistance of the hash function being used.
    pub collision_resistance: u32,
}
```